### ISSUE TITLE
Metrics API return empty response

### ISSUE THREAD

**Issue by pankajvshrma** [pytorch/serve#1961]
### 🐛 Describe the bug

I have written a custom handler. After running Torchserve and registering the workflow, when I try curl http://127.0.0.1:8082/metrics it returns nothing.

### Error logs

http://127.0.0.1:8082/metrics it returns nothing

### Installation instructions

Install Torchserve from source. Not using docker.

### Model Packaing

I use parallel workflow framework and packaged 4 video models and 3 audio models. Two handler files, one for audio and one for video.

### config.properties

enable_envvars_config=true
max_request_size=65535000
number_of_netty_threads=8
netty_client_threads=8
# default_workers_per_model=8
#async_logging=true
job_queue_size=1000
default_response_timeout=300
unregister_model_timeout=300
install_py_dep_per_model=true

inference_address=http://0.0.0.0:8080
management_address=http://0.0.0.0:8081
metrics_address=http://0.0.0.0:8082
enable_metrics_api=true
metrics_format=prometheus

### Versions

------------------------------------------------------------------------------------------
Environment headers
------------------------------------------------------------------------------------------
Torchserve branch: 

torchserve==0.6.0b20221029
torch-model-archiver==0.6.0b20221029

Python version: 3.8 (64-bit runtime)
Python executable: /home/chingari/434/my_env/bin/python

Versions of relevant python libraries:
captum==0.5.0
future==0.18.2
intel-extension-for-pytorch==1.12.300
numpy==1.23.4
nvgpu==0.9.0
psutil==5.9.3
pygit2==1.6.1
pylint==2.6.0
pytest==7.2.0
pytest-cov==4.0.0
pytest-mock==3.10.0
requests==2.28.1
requests-toolbelt==0.10.1
torch==1.12.0+cu113
torch-model-archiver==0.6.0b20221029
torch-workflow-archiver==0.2.4b20221029
torchaudio==0.12.0+cu113
torchserve==0.6.0b20221029
torchtext==0.13.0
torchvision==0.13.0+cu113
transformers==4.11.0
wheel==0.37.1
torch==1.12.0+cu113
torchtext==0.13.0
torchvision==0.13.0+cu113
torchaudio==0.12.0+cu113

Java Version:

OS: Ubuntu 20.04.4 LTS
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: N/A
CMake version: N/A

Is CUDA available: Yes
CUDA runtime version: 11.6.112
GPU models and configuration: 
GPU 0: NVIDIA A10G
Nvidia driver version: 510.85.02
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.4.0

### Repro instructions

torchserve --start \
           --model-store tools/deployment/model_store \
           --workflow-store tools/deployment/workflow_store \
           --ncs \
           --ts-config tools/deployment/config.properties \
           --foreground

**Comment by frankiedrake** [comment]
Do you see metrics in your torchserve logs file (`ts_log.log`, `ts_metrics.log`)?

**Comment by pankajvshrma** [comment]
@frankiedrake yes I see metrics getting logged in torchserve logs file (ts_log.log, ts_metrics.log). Sample log in ts_metrics.log:

**Comment by frankiedrake** [comment]
I was getting the empty response when I specified non-existing metric or when trying to get metrics when no inference API request was made. Maybe logs emit something weird when you're trying to get it?

**Comment by pankajvshrma** [comment]
@frankiedrake yes I tried binding with Prometheus but still getting no output there. Also, I am not using any custom metrics. Even after successful API call I see empty response.

**Comment by pankajvshrma** [comment]
Is it a known problem with workflow based APIs?

**Comment by maaquib** [comment]
Hey @pankajvshrma Right now we only support [3 inference metrics](https://github.com/pytorch/serve/blob/b87d7e59790e06c74f486c186f4a1d47a557b963/frontend/server/src/main/java/org/pytorch/serve/metrics/format/prometheous/PrometheusMetricManager.java#L10-L12) on the prometheus metrics endpoint. Unless you run an inference the metrics will be empty (with a 200OK response). If you have a different expectation, let us know

**Comment by pankajvshrma** [comment]
Hey, metrics endpoint is empty even after inference

**Comment by maaquib** [comment]
@pankajvshrma Seems like a bug with workflow. Will look into this

**Comment by pankajvshrma** [comment]
@maaquib is there any update on this?

**Comment by gasabr** [comment]
I'm encountering the same exact situation in the same environment (0.6.0-cpu torch-serve): even after the model inference /metrics endpoint returns nothing. I can also provide a docker / helm to reproduce

**Comment by hungtooc** [comment]
I'm encountering the same issue

**Comment by Fissium** [comment]
I have the same issue. Any news?

**Comment by Fissium** [comment]
I found a solution: metrics are available if you make inference using models, not a workflow.
E.g. `curl http://127.0.0.1:8080/predictions/dog_breed_wf__dog_breed_classification -T path_to_image/img.jpg` and 
`curl http://127.0.0.1:8080/predictions/dog_breed_wf__cat_dog_classification -T path_to_image/img.jpg`
Then `curl http://127.0.0.1:8082/metrics `returns metrics.

**Comment by pengxin233** [comment]
pytorch/torchserve:0.6.1-gpu I also encountered this problem. Even after calling the model, it still returned empty results.
